kybra/compiler/kybra_generate/src/body/call_global_python_function.rs

```diff
@@ -0,0 +1,58 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+
+pub fn generate() -> TokenStream {
+    quote! {
+        async fn call_global_python_function<'a, T>(
+            function_name: &str,
+            args: impl _KybraTraitIntoFuncArgs
+        ) -> Result<T, String>
+            where
+                for<'b> rustpython::vm::PyObjectRef:
+                    CdkActTryFromVmValue<T, &'b rustpython::vm::VirtualMachine>
+        {
+            let interpreter = unsafe { INTERPRETER_OPTION.as_mut() }
+                .ok_or_else(|| "SystemError: missing python interpreter".to_string())?;
+            let scope = unsafe { SCOPE_OPTION .as_mut() }
+                .ok_or_else(|| "SystemError: missing python scope".to_string())?;
+            let vm = &interpreter.vm;
+            let py_object_ref = scope
+                .globals
+                .get_item(function_name, vm)
+                .map_err(|py_base_exception| py_base_exception.to_rust_err_string(vm))?
+                .call(args, vm)
+                .map_err(|py_base_exception| py_base_exception.to_rust_err_string(vm))?;
+
+            async_result_handler(vm, &py_object_ref, vm.ctx.none())
+                .await
+                .map_err(|py_base_exception| py_base_exception.to_rust_err_string(vm))?
+                .try_from_vm_value(vm)
+                .map_err(|vmc_err| vmc_err.0)
+        }
+
+        fn call_global_python_function_sync<'a, T>(
```

Comment:
Look into getting rid of call_global_python_function_sync

Reply:
You brought up a good point that the `async_result_handler` should be able to handle sync stuff as well so I've just removed this and replaced calls to it with the async version. I'm running tests but if they pass then this will be removed.

Reply:
:grimacing: I don't think we can just collapse the two...
![image](https://github.com/demergent-labs/kybra/assets/5455419/72d91d4e-1b96-4877-9e4c-931ab4425d81)

Reply:
![image](https://github.com/demergent-labs/kybra/assets/5455419/438aea15-9725-4eb1-8ba6-8d9b8c0378b8)